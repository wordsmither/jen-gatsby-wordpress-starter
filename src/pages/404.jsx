import React from 'react'
import { graphql } from 'gatsby'

import Layout from '../components/layout'
import Seo from '../components/seo'

const NotFoundPage = ({ data, location }) => {
	const siteTitle = data.site.siteMetadata.title

	return (
		<Layout location={location} title={siteTitle}>
			<div className="flex flex-col mx-auto text-xl prose text-center mt-12">
				<h2>404: Could not find the requested page</h2>
			</div>
		</Layout>
	)
}

export default NotFoundPage

export function Head() {
	return <Seo title="404: Not Found" />
}

export const pageQuery = graphql`
	query {
		site {
			siteMetadata {
				title
			}
		}
	}
`
