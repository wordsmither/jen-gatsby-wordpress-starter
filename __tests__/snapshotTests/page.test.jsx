import * as Gatsby from 'gatsby'
import { render } from '@testing-library/react'

import PageTemplate from '../../src/templates/page'
import pages from '../data/pages.json'

const StaticQuery = vi.spyOn(Gatsby, 'StaticQuery')
const page = pages[0]
const next = pages[1]
const previous = null
describe('<PageTemplate />', () => {
	beforeEach(() => {
		StaticQuery.mockImplementation(() => ({ data: { page } }))
	})

	it('should render with a page', () => {
		const { asFragment } = render(
			<PageTemplate data={page} pageContext={{ next, previous }} />,
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
