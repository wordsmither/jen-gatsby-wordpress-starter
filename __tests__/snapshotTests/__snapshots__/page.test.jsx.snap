// Vitest Snapshot v1

exports[`<PageTemplate /> > should render with a page 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col"
  >
    <div
      class="my-0 pt-10 px-5 text-xl"
    >
      <nav>
        <ul
          class="flex flex-row flex-wrap sm:flex-nowrap list-none justify-between max-w-screen-sm mx-auto"
        >
          <li
            class="mr-auto"
          >
            <a
              class="font-sans hover:underline"
              href="/"
            >
              🏠 Home
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/posts"
            >
              📰 Posts
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/pages"
            >
              📑 Pages
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/examples"
            >
              ⚛️ Examples
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <main
      class="mb-auto"
    >
      <article
        class="prose lg:prose-xl mt-10 mx-auto"
      >
        <h1>
          Test Page
        </h1>
        <p
          class="text-sm text-gray-600"
        >
          Invalid Date
        </p>
        <a
          class="font-normal"
          href="/pages"
        >
          Pages →
        </a>
        <div
          class="mt-12 max-w-lg mx-auto lg:grid-cols-3 lg:max-w-screen-lg"
        />
        <p>
          Sorry, no page data was found at this route.
        </p>
        <hr
          class="mt-10"
        />
        <nav
          class="flex flex-wrap px-6"
        >
          <a
            class="underline font-medium ml-auto"
            href="/pagesundefined"
          >
             →
          </a>
        </nav>
      </article>
    </main>
    <footer
      class="w-full text-white bg-black p-4 mt-12"
    >
      <nav
        class="flex flex-col max-w-lg mx-auto lg:max-w-screen-lg"
      >
        <ul>
          <li
            class="ml-3 list-disc text-blue-300 hover:text-blue-100"
          >
            <a
              class="text-blue-300 hover:text-blue-100 focus:text-purple-600 active:text-purple-300 "
              href="/posts/example-post-with-image/"
            >
              Example Post with Image
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="flex my-4 p-2"
      >
        <span
          class="mx-auto"
        >
          © 2023 Built with 
          <a
            class="text-purple-300 underline hover:text-blue-100"
            href="https://www.gatsbyjs.com"
          >
            Gatsby
          </a>
           and 
          <a
            class="text-blue-200 underline hover:text-blue-100"
            href="https://wordpress.org/"
          >
            WordPress
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
